ioc_copier/ioc_copier.py

```diff
@@ -156,9 +191,21 @@ def get_file_text(file, ioc, root):
     :return:
         The text from the file, with the IOC number updated to the current copy.
     """
-    with open(os.path.join(root, file), "r") as file_pointer:
+    path = os.path.join(root, file)
+    with open(path, "r") as file_pointer:
         text = file_pointer.readlines()
+    final_lines = []
+    if START_COPY == 1:
+        if file == "st.cmd":
+            final_lines = [text.pop(-1)]
+        elif file == "config.xml":
+            return generate_config(ioc)
+        elif path.endswith(r"App\src\Makefile"):
+            final_lines = [text.pop(-3), text.pop(-2), text.pop(-1)]
```

Comment:
`-3`, `-2` and `-1` are magic numbers.

Also taking the last 3 lines of the file seems very error-prone, is there not a better way to detect lines we shouldn't change (like "don't change a line that looks like `include $(TOP)/AMINT2L-IOC-01App/src/build.mak`" for example)

Reply:
Even if we do have to do this can't this just be `final_lines = text[-3:]`?

Reply:
replaced with filter to find lines discussing build.mak and them to list of indexes to skip